src/core/math.rs

```diff
@@ -43,3 +43,15 @@ pub fn div(data: Data) -> Result<Data, String> {
 
     return Ok(result);
 }
+
+/// remainder of left operand by right operand division.
+/// Raises a runtime error if there is a division by zero.
+pub fn remainder(data: Data) -> Result<Data, String> {
+    let result = match binop(data) {
+        (Data::Real(_),   Data::Real(r)) if r == 0.0 => Err("Division by zero")?,
+        (Data::Real(l),   Data::Real(r)) => Data::Real(l % r),
```

Comment:
So here we're using Rust's remainder between floats. There are multiple different ways to do remainders on negative numbers, so I guess we need to figure out whether we prefer the current way remainders are done for negative numbers in Rust or `rem_euclid`, which calculates the least nonnegative remainder?

Reply:
I don't have any preferences about that right now. problems with this usually occur when the user expects some behavior (usually coming from a different language) and then gets some different behavior. I usually opt for the mathematical convention in these cases, but I'm not sure what it is for modulus operations... I'll try to think about it today, and we'll see. if you have some preference we could just use that, I don't have strong opinions on this.

Reply:
I'm pretty sure the mathematical convention for modulus operators is the euclidean remainder, which is always positive. I was a bit confused with rust's default remainder behavior when I first came across it. Let me know what you think.

Reply:
Sure, I'll change it to use `rem_euclid`